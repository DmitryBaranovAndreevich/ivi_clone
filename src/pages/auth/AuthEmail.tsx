import styles from '../login/login.module.scss';
import Line from '../../components/login/Line/Line';
import LoginHeader from '../../components/login/LoginHeader/LoginHeader';
import LoginLayout from '../../components/login/LoginLayout/LoginLayout';
import LoginMessage from '../../components/login/LoginMessage/LoginMessage';
import Input from '../../components/UI/Input/Input';
import useForm from '../../hooks/useForm';
import { FormEvent, useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import ErrorMessage from '../../components/login/ErrorMessage/ErrorMessage';
import * as Yup from 'yup';
import { userAuthSlice } from '../../store/reducers/UserAuthSlice';
import { useAppDispatch } from '../../hooks/redux';
import { useTranslation } from 'react-i18next';
import RedButton from '../../components/UI/redButton/RedButton';

const AuthEmail = () => {
  const { t } = useTranslation();
  const { setEmailNameSurName } = userAuthSlice.actions;
  const dispatch = useAppDispatch();
  const { value, handleChange } = useForm({ email: '', first_name: '', second_name: '' });
  const [error, setError] = useState('');
  const navigate = useNavigate();

  const shema = Yup.object().shape({
    email: Yup.string()
      .email(t('auth.dataErr') as string)
      .required(t('auth.emailErr') as string),
    first_name: Yup.string().required(t('auth.nameErr') as string),
    second_name: Yup.string().required(t('auth.secNameErr') as string),
  });

  useEffect(() => {
    setError('');
  }, [value]);

  const handleClick = (e: FormEvent) => {
    e.preventDefault();
    shema
      .validate(value)
      .then((res) => {
        dispatch(setEmailNameSurName(res));
        navigate('/auth/password');
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  return (
    <LoginLayout>
      <LoginHeader title={t('auth.register')} />
      <Line persent={'10%'} />
      <form className={styles.wrapper}>
        <LoginMessage>
          <span className={styles.span}>{t('auth.register1')}</span> {t('login.use')}
        </LoginMessage>
        <Input
          placeholder={t('auth.emailEnt')}
          name={'email'}
          value={value.email}
          onChange={handleChange}
        />
        <Input
          placeholder={t('auth.nameEnt')}
          name={'first_name'}
          value={value.first_name}
          onChange={handleChange}
        />
        <Input
          placeholder={t('auth.secNameEnt')}
          name={'second_name'}
          value={value.second_name}
          onChange={handleChange}
        />
        <RedButton
          addingClass={styles.button}
          text={t('login.continue')}
          type={'submit'}
          onClick={handleClick}
        />
        {error && <ErrorMessage error={error} />}
      </form>
    </LoginLayout>
  );
};

export default AuthEmail;
