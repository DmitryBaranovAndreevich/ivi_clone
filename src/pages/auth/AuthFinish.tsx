import { useNavigate } from 'react-router-dom';
import Line from '../../components/login/Line/Line';
import LoginHeader from '../../components/login/LoginHeader/LoginHeader';
import LoginLayout from '../../components/login/LoginLayout/LoginLayout';
import LoginMessage from '../../components/login/LoginMessage/LoginMessage';
import styles from '../login/login.module.scss';
import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';

const AuthFinish = () => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  useEffect(() => {
    const redirect = () => {
      navigate('/', { replace: true });
    };

    const timer = setTimeout(redirect, 4500);
    return () => {
      clearTimeout(timer);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <LoginLayout>
      <LoginHeader title={t('auth.congratulations')} />
      <Line persent={'100%'} />
      <LoginMessage>{t('auth.register1')}</LoginMessage>
      <div className={`${styles.div} ${styles.animation} ${styles.animationTime2}`}>
        <LoginMessage success>{t('login.finish')}</LoginMessage>
      </div>
    </LoginLayout>
  );
};

export default AuthFinish;
