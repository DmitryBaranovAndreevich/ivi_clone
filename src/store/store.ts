import reviewReducer from './reducers/ReviewSlice';
import moviesSortReduser from './reducers/MoviesSort';
import { appApi } from './api/appApi';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import userLoginReduser from './reducers/UserLoginSlice';
import appReducer from './reducers/App';
import moviesFilterReduser from './reducers/MoviesFilter';
import { filmApi } from './api/filmApi';
import { personApi } from './api/personApi';
import { reviewApi } from './api/reviewApi';

const rootReducer = combineReducers({
  appReducer,
  userLoginReduser,
  moviesSortReduser,
  reviewReducer,
  moviesFilterReduser,
  [appApi.reducerPath]: appApi.reducer,
  [filmApi.reducerPath]: filmApi.reducer,
  [personApi.reducerPath]: personApi.reducer,
  [reviewApi.reducerPath]: reviewApi.reducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat([appApi.middleware, filmApi.middleware, personApi.middleware]),
  });
};

export type TRootState = ReturnType<typeof rootReducer>;
export type TAppStore = ReturnType<typeof setupStore>;
export type TAppDispatch = TAppStore['dispatch'];
