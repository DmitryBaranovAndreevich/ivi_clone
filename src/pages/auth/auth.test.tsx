import { Provider } from 'react-redux';
import { store } from '../../store/mainStore';
import AuthEmail from './AuthEmail';
import AuthFinish from './AuthFinish';
import renderer from 'react-test-renderer';
import * as router from 'react-router';
import { MemoryRouter } from 'react-router-dom';

describe('Тестим страницу авторизации', () => {
  const mockUseLocationValue = {
    key: 'key',
    pathname: '/testroute',
    search: '',
    hash: '',
    state: null,
  };
  const navigate = jest.fn();

  jest.mock('react-i18next', () => ({
    useTranslation: jest.fn(),
  }));

  beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    jest.spyOn(router, 'useLocation').mockImplementation(() => mockUseLocationValue);
  });
  it('Страница authEmail рендерится без ошибок', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AuthEmail />
        </MemoryRouter>
      </Provider>
    );
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
  it('Страница authFinish рендерится без ошибок', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AuthFinish />
        </MemoryRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
