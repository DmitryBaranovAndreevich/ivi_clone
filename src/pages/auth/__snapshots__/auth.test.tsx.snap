// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим страницу авторизации Страница authEmail рендерится без ошибок 1`] = `
<div>
  <div>
    <p>
      auth.register
    </p>
    <button
      className="undefined undefined"
      onClick={[Function]}
    >
      <img
        alt="cross"
        src={{}}
      />
    </button>
  </div>
  <a
    href="/auth/email"
    onClick={[Function]}
    style={
      {
        "textDecoration": "none",
      }
    }
  >
    <p>
      login.register
    </p>
  </a>
  <div
    style={
      {
        "--size": "10%",
      }
    }
  />
  <form>
    <p
      style={{}}
    >
      <span>
        auth.register1
      </span>
       
      login.use
    </p>
    <div>
      <span>
        auth.emailEnt
      </span>
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <span>
        auth.nameEnt
      </span>
      <input
        name="first_name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <span>
        auth.secNameEnt
      </span>
      <input
        name="second_name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      className="undefined "
      onClick={[Function]}
      type="submit"
    >
      login.continue
    </button>
  </form>
</div>
`;

exports[`Тестим страницу авторизации Страница authFinish рендерится без ошибок 1`] = `
<div>
  <div>
    <p>
      auth.congratulations
    </p>
    <button
      className="undefined undefined"
      onClick={[Function]}
    >
      <img
        alt="cross"
        src={{}}
      />
    </button>
  </div>
  <a
    href="/auth/email"
    onClick={[Function]}
    style={
      {
        "textDecoration": "none",
      }
    }
  >
    <p>
      login.register
    </p>
  </a>
  <div
    style={
      {
        "--size": "100%",
      }
    }
  />
  <p
    style={{}}
  >
    auth.register1
  </p>
  <div
    className="undefined undefined undefined"
  >
    <p
      style={
        {
          "backgroundColor": "lightgreen",
          "color": "black",
        }
      }
    >
      login.finish
    </p>
  </div>
</div>
`;
